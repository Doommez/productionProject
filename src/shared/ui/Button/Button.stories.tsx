import type { Meta, StoryObj } from '@storybook/react-webpack5';
import 'app/styles/index.scss';

import { ThemeDecorator } from 'shared/config/storybook/ThemeDecorator/ThemeDecorator';
import { Theme } from 'app/providers/themeProvider';
import { Button, ButtonSize, ThemeButton } from './Button';

const meta = {
    title: 'shared/Button',
    component: Button,
    parameters: {
        layout: 'centered'
    },
    tags: ['autodocs'],
    argTypes: {
        themeButton: {
            control: 'select',
            options: Object.values(ThemeButton)
        },
        children: {
            control: 'text'
        },
        onClick: { action: 'clicked' },
        size: {
            control: 'select',
            options: Object.values(ButtonSize)
        },
        square: {
            control: 'select',
            options: [true, false]
        }

    },
    decorators: [ThemeDecorator(Theme.LIGHT)]
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        children: 'кнопка'
    }
};

export const Clear: Story = {
    args: {
        children: 'кнопка',
        themeButton: ThemeButton.CLEAR
    }
};
export const Clicked: Story = {
    args: {
        children: 'кнопка'

    }
};
export const Outline: Story = {
    args: {
        children: 'кнопка',
        themeButton: ThemeButton.OUTLINE
    }
};

export const Size: Story = {
    args: {
        children: 'кнопка',
        size: ButtonSize.XL
    }
};
