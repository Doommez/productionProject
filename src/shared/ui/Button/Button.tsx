import { ButtonHTMLAttributes, FC } from 'react';
import { classNames } from 'shared/lib/classNames/classNames';
import cls from './Button.module.scss';

export enum ThemeButton {
    CLEAR = 'clear',
    OUTLINE = 'outline',
    CLEAR_INVERTED = 'clearInverted',
    BACKGROUND = 'background',
    BACKGROUND_INVERTED = 'backgroundInverted',
}

export enum ButtonSize {
    L = 'size_l',
    M = 'size_m',
    XL = 'size_xl',
}

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    className?: string;
    themeButton?: ThemeButton;
    square?: boolean;
    size?: ButtonSize;
}

export const Button: FC<ButtonProps> = (props) => {
    const {
        className,
        themeButton,
        children,
        square,
        size,
        ...otherProps
    } = props;

    const mods: Record<string, boolean> = {
        [cls[themeButton]]: true,
        [cls.square]: square,
        [cls[size]]: true
    };

    return (
        <button
            type="button"
            className={classNames(cls.Button, mods, [
                className
            ])}
            {...otherProps}
        >
            {children}
        </button>
    );
};
