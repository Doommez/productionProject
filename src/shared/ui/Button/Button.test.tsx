import { render, screen } from '@testing-library/react';
import { Button } from 'shared/ui/Button/Button';

describe('Button', () => {
    test('Button render text test', () => {
        const textButton = 'test';
        render(<Button>{textButton}</Button>);
        expect(screen.getByText(textButton))
            .tobei;
    });
});
