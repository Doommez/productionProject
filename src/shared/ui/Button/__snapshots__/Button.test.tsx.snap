// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot test Button snapshot 1`] = `
<div>
  <div
    class="app light"
  >
    <button
      class="Button undefined"
      type="button"
    >
      кнопка
    </button>
  </div>
</div>
`;
